contracts/HybridCustody.cdc

```diff
@@ -979,22 +1029,17 @@ pub contract HybridCustody {
         /// mechanism intended to easily transfer 'root' access on this account to another account and an attempt to
         /// minimize access vectors.
         ///
-        pub fun giveOwnership(to: Address) {
+        access(Owner) fun giveOwnership(to: Address) {
             self.seal()
             
             let acct = self.borrowAccount()
-            // Unlink existing owner's Capability if owner exists
-            if self.acctOwner != nil {
-                acct.unlink(
-                    PrivatePath(identifier: HybridCustody.getOwnerIdentifier(self.acctOwner!))!
-                )
-            }
+
             // Link a Capability for the new owner, retrieve & publish
             let identifier =  HybridCustody.getOwnerIdentifier(to)
-            let cap = acct.link<&{OwnedAccountPrivate, OwnedAccountPublic, MetadataViews.Resolver}>(
-                    PrivatePath(identifier: identifier)!,
-                    target: HybridCustody.OwnedAccountStoragePath
-                ) ?? panic("failed to link child account capability")
+            let cap = acct.capabilities.storage.issue<auth(Owner) &{OwnedAccountPrivate, OwnedAccountPublic, ViewResolver.Resolver}>(HybridCustody.OwnedAccountStoragePath)
+
+            // make sure we can borrow the newly issued owned account
+            cap.borrow()!.borrowAccount()
```

Comment:
This is just so that a failure message is more clear, I've been rebuilding my habits to use panic where it makes sense instead of force unwrap so that I can send a more readable failure along